tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1560,6 +1568,48 @@ bool StorageManager::is_vacuum_file(const URI& uri) const {
   return false;
 }
 
+Status StorageManager::get_array_schema_uris(
+    const URI& array_uri, std::vector<URI>* uris) const {
+  STATS_START_TIMER(stats::GlobalStats::TimerType::READ_GET_ARRAY_SCHEMAS)
+
+  const URI schema_uri =
+      array_uri.join_path(constants::array_schema_folder_name);
+  bool is_dir = false;
+  RETURN_NOT_OK(vfs_->is_dir(schema_uri, &is_dir));
```

Comment:
Could we avoid this by checking the result of `ls` below. In general, note that every `ls`, `is_file` and `is_dir` can be very expensive on cloud object stores, so we need to minimize them as much as we can. cc @Shelnutt2

Reply:
Yes, we can avoid using them. when getting all schem uris vector,  I will add the schema  file(array_name/__array_schema.tdb) as first schema and treat it timestamp as 0.